test/gtest/uct/ib/test_ib_md.cc

```diff
@@ -18,26 +18,49 @@
 class test_ib_md : public test_md
 {
 protected:
-    void ib_md_umr_check(void *rkey_buffer,
-                         bool amo_access,
-                         size_t size = 8192);
+    void init() override;
+    uct_ib_md_t const &ib_md() const;
+    void ib_md_umr_check(void *rkey_buffer, bool amo_access,
+                         size_t size = 8192, int aligned = 0);
     bool has_ksm() const;
-    bool check_umr(uct_ib_md_t *ib_md) const;
+    bool check_umr() const;
+
+private:
+    uint32_t m_mlx5_flags;
 };
 
+void test_ib_md::init() {
+    test_md::init();
+
+    /* Save mlx5 IB md flags because failed atomic registration will modify it */
+    if (ib_md().dev.flags & UCT_IB_DEVICE_FLAG_MLX5_PRM) {
+        m_mlx5_flags = ucs_derived_of(md(), uct_ib_mlx5_md_t)->flags;
+    } else {
+        m_mlx5_flags = 0;
+    }
+}
+
+uct_ib_md_t const &test_ib_md::ib_md() const {
+    return *ucs_derived_of(md(), uct_ib_md_t);
+}
 
 /*
  * Test that ib md does not create umr region if
  * UCT_MD_MEM_ACCESS_REMOTE_ATOMIC is not set
  */
-
-void test_ib_md::ib_md_umr_check(void *rkey_buffer,
-                                 bool amo_access,
-                                 size_t size)
+void test_ib_md::ib_md_umr_check(void *rkey_buffer, bool amo_access,
+                                 size_t size, int aligned)
 {
     ucs_status_t status;
     size_t alloc_size;
     void *buffer;
+    int ret;
+
+    if (amo_access &&
+        ((IBV_DEV_ATTR(&ib_md().dev, vendor_part_id) == 4117) ||   /* CX4 */
```

Comment:
Do we need to include any other devices here (see `uct_ib_builtin_device_specs`) ?